services/ossf-scorecard/ossf-scorecard.service.js

```diff
@@ -0,0 +1,59 @@
+import Joi from 'joi'
+import { BaseJsonService } from '../index.js'
+import { colorScale } from '../color-formatters.js'
+
+const schema = Joi.object({
+  score: Joi.required(),
+}).required()
+
+const ossfScorecardColorScale = colorScale(
+  [2, 5, 8, 10],
+  ['red', 'yellow', 'yellowgreen', 'green', 'brightgreen']
+)
+
+export default class OSSFScorecard extends BaseJsonService {
+  static category = 'analysis'
+
+  static route = { base: 'ossf-scorecard', pattern: ':host/:orgName/:repoName' }
+
+  static examples = [
+    {
+      title: 'OSSF-Scorecard Score',
+      namedParams: {
+        host: 'github.com',
+        orgName: 'rohankh532',
+        repoName: 'org-workflow-add',
+      },
+      staticPreview: this.render({ score: '7.5' }),
+    },
+  ]
+
+  static defaultBadgeData = { label: 'score' }
+
+  static render({ score }) {
+    return {
+      message: score,
+      color: ossfScorecardColorScale(score),
+    }
+  }
+
+  async fetch({ host, orgName, repoName }) {
+    return this._requestJson({
+      schema,
+      url: `https://api.securityscorecards.dev/projects/${host}/${orgName}/${repoName}`,
+      errorMessages: {
+        404: 'invalid repo path',
+        500: 'upstream service error',
```

Comment:
Our request handling infrastructure actually covers these cases already. We obviously allow for individual classes to apply their own message overriding for specific response codes, but it's not necessary to do so. In this case since we're not doing anything particularly unique/adding additional context I'd recommend that we remove all of these overrides, but at a minimum do so for the 500 (in order to keep it consistent and make things a tad easier/more obvious for us from an operations/support perspective)

Reply:
I removed the 500 and kept the 404